app/firebase/util.ts

```diff
@@ -61,44 +77,131 @@ async function getSchoolList() {
  * @param callback The function to call with the updated data.
  * @returns A function to unsubscribe from the real-time updates.
  */
-function listenToSchoolDirections(schoolName, callback) {
-  // Generate a database reference specifically targeting the requested school's directions.
+export function listenToSchoolDirections(
+  schoolName: string,
+  callback: (data: SchoolDirections | null) => void
+): () => void {
   const schoolRef = ref(database, `/SchoolDirections/${schoolName}`);
 
-  // Attach a listener to get real-time updates.
   const unsubscribe = onValue(
     schoolRef,
     (snapshot) => {
       if (snapshot.exists()) {
-        // If data exists, call the callback with the entire data object for this school.
         callback(snapshot.val());
       } else {
         console.log('No data available for:', schoolName);
-        callback(null); // Call the callback with null if no data is found.
+        callback(null);
       }
     },
     (error) => {
       console.error('Error fetching school directions:', error);
-      callback(null); // Call the callback with null to indicate an error occurred.
+      callback(null);
     }
   );
 
-  // Return a function to unsubscribe from the listener when it's no longer needed.
   return () => off(schoolRef, 'value', unsubscribe);
 }
 
-// Save function for school transportation details
-async function updateSchoolDirections(schoolName: string, field: string, value: string) {
-  // Generate a database reference specifically targeting the requested school's directions.
+/**
+ * Updates a specific field in the school directions for a given school.
+ * @param schoolName The name of the school.
+ * @param field The specific field to update.
+ * @param value The new value for the field.
+ * @returns A promise resolving to true if the update is successful, otherwise null.
+ */
+export async function updateSchoolDirections(
+  schoolName: string,
+  field: string,
+  value: string
+): Promise<boolean | null> {
   const schoolRef = ref(database, `/SchoolDirections/${schoolName}/${field}`);
   try {
-    set(schoolRef, value);
+    await set(schoolRef, value);
     return true;
   } catch (error) {
     console.error('Error updating school directions: ', error);
     return null;
   }
 }
 
-// Export the functions for use in other parts of the application.
-export { getSchoolList, listenToSchoolDirections, updateSchoolDirections };
+/**
```

Comment:
We don't really need this since they wouldn't be changing it but I guess we can just keep it here